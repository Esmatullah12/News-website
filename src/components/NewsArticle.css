.newsArticle {
  width: 300px;
  border: 0.5px solid rgb(57, 57, 57);
  box-shadow: rgba(60, 64, 67, 0.3) 0 1px 2px 0, rgba(60, 64, 67, 0.15) 0 1px 3px 1px;
  background-color: #f0f0f0;
}

.newsArticle:hover {
  transform: scale(1.05);
  transition: transform 0.5s;
}

.new-card-img {
  /* border-radius: 7px 7px 0 0; */
  width: 100%;
  height: 250px;
  object-fit: cover;
  border-bottom: 1px solid black;
}

.news-info {
  height: 200px;
  padding: 20px;
  overflow: hidden;
}

.news-title {
  font-size: 1.2rem;
  font-weight: 600;
  max-width: 80%;
}

.news-date {
  font-size: 2rem;
}

.read-more {
  display: flex;
  align-self: flex-end;
}
