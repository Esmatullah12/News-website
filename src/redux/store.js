import { configureStore } from '@reduxjs/toolkit';
import newsSlice from './news/newsSlice';

const store = configureStore({
  reducer: {
    news: newsSlice,
  },
});

export default store;
