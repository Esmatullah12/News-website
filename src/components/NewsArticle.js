import React from 'react';
import PropTypes from 'prop-types';
import './NewsArticle.css';

// const truncateTitle = (title) => {
//   const words = title.split(' ');
//   if (words.length > 14) {
//     return `${words.slice(0, 5).join(' ')}...`;
//   }
//   return title;
// };

const NewsArticle = ({
  date, title, imageUrl, newsId,
}) => {
  const options = {
    month: 'numeric',
    day: 'numeric',
    weekday: 'long',
  };
  const formattedDate = new Date(date).toLocaleDateString(undefined, options);
  const [weekday, month, day] = formattedDate.split(',');

  return (
    <div key={newsId} className="newsArticle">
      <img className="new-card-img" src={imageUrl} alt="News Article" />
      <div className="news-info">
        <p className="news-date">
          {day}
          {month}
        </p>
        <p>{weekday}</p>
        <p className="news-title">{title}</p>
        <p className="read-more">Read more...</p>
      </div>
    </div>
  );
};

NewsArticle.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  newsId: PropTypes.number.isRequired,
};

export default NewsArticle;
