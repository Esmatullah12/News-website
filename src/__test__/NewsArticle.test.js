import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import NewsArticle from '../components/NewsArticle';
import store from '../redux/store';

describe('Test NewsArticle component', () => {
  it('it should render NewsArticle component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <NewsArticle />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
