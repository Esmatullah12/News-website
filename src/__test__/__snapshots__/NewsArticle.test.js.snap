// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NewsArticle component it should render NewsArticle component correctly 1`] = `
<div>
  <div
    class="newsArticle"
  >
    <img
      alt="News Article"
      class="new-card-img"
    />
    <div
      class="news-info"
    >
      <p
        class="news-date"
      />
      <p>
        Invalid Date
      </p>
      <p
        class="news-title"
      />
      <p
        class="read-more"
      >
        Read more...
      </p>
    </div>
  </div>
</div>
`;
